benchmark/benchmark_gemm_split_k_fp8_e4m3.py

```diff
@@ -0,0 +1,184 @@
+            args={"mode": "full", "dtype": torch.float32},
+        ),
+    ]
+)
+def bench_speed_gemm_split_k_fp8(m, k, n, provider, mode, dtype, device="cuda"):
+    a_fp8 = torch.randn((m, k), device=device, dtype=dtype).to(torch.float8_e4m3fn)
+    b_fp8 = torch.randn((k, n), device=device, dtype=dtype).to(torch.float8_e4m3fn)
+
+    a_float = a_fp8.float().requires_grad_()
+    b_float = b_fp8.float().requires_grad_()
+
+    def fwd_liger():
+        return LigerFP8GemmSplitKFunction.apply(a_fp8, b_fp8)
+
+    def fwd_torch():
+        return torch.matmul(a_float, b_float)
+
+    fwd_torch_compiled = torch.compile(fwd_torch)
+
+    if provider == "liger":
+        fwd_fn = fwd_liger
+    elif provider == "torch":
+        fwd_fn = fwd_torch
+    else:
+        fwd_fn = fwd_torch_compiled
+
+    quantiles = [0.5, 0.2, 0.8]
+
+    if mode == "forward":
+        ms, min_ms, max_ms = triton.testing.do_bench(fwd_fn, quantiles=quantiles)
+    elif mode == "full":
+
+        def full():
+            y = fwd_fn()
+            if provider == "liger":
+                # compute manually gradients for Liger to avoid: "ufunc_add_CUDA" not implemented for 'Float8_e4m3fn'
+                dc = torch.ones_like(y, dtype=torch.float8_e4m3fn)
+                LigerFP8GemmSplitKFunction.apply(dc, b_fp8.t())
+                LigerFP8GemmSplitKFunction.apply(a_fp8.t(), dc)
+            else:
+                torch.sum(y).backward()
+
+        ms, min_ms, max_ms = triton.testing.do_bench(full, quantiles=quantiles)
+
+    return ms, min_ms, max_ms
+
+
+def benchmark_speed_gemm_split_k_fp8_wrapper():
+    _print_speed_banner()
+
+    curr_dir = get_current_file_directory()
+    dir_name = "gemm_split_k_fp8_speed"
+    output_dir = os.path.join(curr_dir, dir_name)
+    os.makedirs(output_dir, exist_ok=True)
+
+    bench_speed_gemm_split_k_fp8.run(save_path=output_dir, print_data=True)
+
+
+@triton.testing.perf_report(
+    [
+        triton.testing.Benchmark(
+            x_names=["m", "k", "n"],
+            x_vals=[
+                (64, 64, 64),
+                (256, 256, 256),
+                (512, 512, 512),
+                (1024, 1024, 1024),
+                (64, 128, 64),
+                (256, 512, 256),
+                (512, 1024, 512),
+                (1024, 2048, 1024),
+            ],
+            xlabel="Matrix Size (m x k x n)",
+            line_arg="provider",
+            line_vals=["liger", "torch", "torch_compile"],
+            line_names=["Liger", "PyTorch", "Torch Compile"],
+            styles=[("blue", "solid"), ("orange", "solid"), ("green", "solid")],
+            ylabel="GPU memory usage (MB)",
+            plot_name="gemm-split-k-fp8-memory-benchmark",
+            args={"dtype": torch.float32},
+        )
+    ]
+)
+def bench_memory_gemm_split_k_fp8(m, k, n, provider, dtype, device="cuda"):
+    a_fp8 = torch.randn((m, k), device=device, dtype=dtype).to(torch.float8_e4m3fn)
```

Comment:
ditto. Let's try to create bf16 input and compare the speed/memory of torch._scaled_mm v.s. the fp8 kernel and then compare the joint time of quant + dequant + matmul (with fp8 scaled factor) Thanks!